Gems/AzQtComponentsForPython/Code/CMakeLists.txt

```diff
@@ -0,0 +1,146 @@
+
+include(${pal_dir}/PAL_${PAL_PLATFORM_NAME_LOWERCASE}.cmake)
+
+if (PAL_TRAIT_AZQTCOMPONENTSFORPYTHON_SUPPORTED)
+    # The name of the Python import module we're generating.
+    set(BINDINGS_LIBRARY "azqtpyside")
+
+    # Header file containing all the types/classes we're exposing to Python.
+    set(wrapped_header ${CMAKE_CURRENT_LIST_DIR}/bindings_azqtcomponents.h)
+
+    # Typesystem xml.
+    set(typesystem_file ${CMAKE_CURRENT_LIST_DIR}/bindings_azqtcomponents.xml)
+
+    include(${CMAKE_CURRENT_LIST_DIR}/wrapped_files.cmake)
+    
+    # Allow AUTOMOC/AUTOUIC on generated files.
+    if(POLICY CMP0071)
+      cmake_policy(SET CMP0071 NEW)
+    endif()
+
+    project(AzQtComponent_Bindings LANGUAGES CXX)
+    set(CMAKE_AUTOMOC ON)
+    find_package(Qt5 5.12 REQUIRED COMPONENTS Core Gui Widgets)
+                 
+    # The library we're creating bindings for.
+    set(AZQTCOMPONENTS_LIBRARY "AzQtComponents")
+
+    # Check we have a valid version of Python.
+    if(NOT python_interpreter)
+        find_program(python_interpreter "python")
+    endif()
+
+    if(NOT python_interpreter)
+        message(FATAL_ERROR "Error: Failed to locate Python installation.")
+    endif()
+    
+    # Load the library/exe locations. These are platform specific.
+    include(${pal_dir}/azqtcomponentsforpython_file_settings.cmake)
+    
+    #Define the shared library for which we will create bindings.
+    set_property(TARGET ${AZQTCOMPONENTS_LIBRARY} PROPERTY PREFIX "")
+
+    # Ensure the generated wrapper files have the correct header.
+    set(LICENSE_FILE ${CMAKE_CURRENT_LIST_DIR}/license_header.txt)
+
+    # Set up the options to pass to shiboken.
+    set(shiboken_options --generator-set=shiboken --enable-parent-ctor-heuristic
+        --enable-pyside-extensions --enable-return-value-heuristic --use-isnull-as-nb_nonzero
+        --avoid-protected-hack --language-level=c++17 --debug-level=full
+        --license-file=${LICENSE_FILE}
+        -I${LY_ROOT_FOLDER}/Code/Framework/AzCore
+        -I${LY_ROOT_FOLDER}/Code/Framework/AzQtComponents
+        -I${LY_ROOT_FOLDER}/Code/Framework/AzCore/Platform/${PAL_PLATFORM_NAME} 
+        -I${QT_MODULE_DIR}/include
+        -I${QT_MODULE_DIR}/include/QtWidgets
+        -I${QT_MODULE_DIR}/include/QtCore
+        -I${QT_MODULE_DIR}/include/QtGui
+        -T${PYSIDE2_TYPESYSTEM_PATH}
+        -T${PYSIDE2_SHARE_PATH}
+        --output-directory=${CMAKE_CURRENT_BINARY_DIR}
+    )
+
+    set(generated_sources_dependencies ${wrapped_header} ${typesystem_file})
+
+    add_custom_command(
+        OUTPUT ${WRAPPED_FILES}
+        COMMAND ${SHIBOKEN_GENERATOR_EXE_PATH} ${shiboken_options} ${wrapped_header} ${typesystem_file}
+        DEPENDS ${generated_sources_dependencies} 
+        COMMENT "Running generator for ${typesystem_file}."
+        VERBATIM
+    )
+
+    # Set up the project for building the code generated by shiboken.
+    ly_add_target(
+        NAME ${GEM_NAME} GEM_MODULE
```

Comment:
We should probably call this `${GEM_NAME}.Editor` as what we aim for most is consistency with other gems in the code base when it comes to CMake logic and structure.

Reply:
Changed.